scripts/check_camera_sharpness.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+"""Show live image of a Pylon camera and indicate whether it passes the sharpness test.
+
+The sharpness of the image is evaluated by running Canny edge detection on the image and
+computing the mean value of the resulting edge image.  A higher value means more edges
+which indicates a sharper image.
+
+The parameters can be adjusted via sliders in the GUI (note that the sliders in the GUI
+only support integer values).
+
+This test is also performed in robot_fingers/trifingerpro_post_submission.py.
+"""
+import argparse
+import dataclasses
+import typing
+from collections import deque
+
+import cv2
+import numpy as np
+
+import trifinger_cameras
+from trifinger_cameras import utils
+
+
+def add_label(
+    image: np.ndarray, label: str, position: typing.Literal["top", "bottom"]
+):
+    """Add label to the given image.
+
+    Args:
+        image: The image.
+        label: Text to be added.
+        position: Where to add the label (either "top" or "bottom").
+
+    Returns:
+        Image with added label.
+    """
+    label_height = 20
+
+    if position == "top":
+        offset = 0
+    elif position == "bottom":
+        offset = image.shape[0] - label_height
+    else:
+        ValueError(f"Invalid position '{position}'")
+
+    # darken label background
+    row_slice = slice(offset, (offset + label_height))
+    image[row_slice, :, :] = 0.3 * image[row_slice, :, :]
+
+    image = cv2.putText(
+        image,
+        label,
+        (5, offset + label_height - 5),
+        cv2.FONT_HERSHEY_SIMPLEX,
+        0.5,
+        (255, 255, 255),
+    )
+    return image
+
+
+def auto_canny_params(
+    image: np.ndarray, sigma: float = 0.33
+) -> typing.Tuple[float, float]:
+    median = np.median(image)
+    lower = max(0, (1.0 - sigma) * median)
+    upper = min(255, (1.0 + sigma) * median)
+
+    return lower, upper
+
+
+@dataclasses.dataclass
+class Params:
+    canny_threshold1: float
+    canny_threshold2: float
+    edge_mean_threshold: float
+
+    def set_canny_threshold1(self, value: float) -> None:
+        self.canny_threshold1 = value
+
+    def set_canny_threshold2(self, value: float) -> None:
+        self.canny_threshold2 = value
+
+    def set_edge_mean_threshold(self, value: float) -> None:
+        self.edge_mean_threshold = value
+
+
+def main():
+    argparser = argparse.ArgumentParser(description=__doc__)
+    argparser.add_argument(
+        "camera_id",
+        type=str,
+        help="""ID of the camera that is used.  If --pylon is set this refers
+            to the DeviceUserId, otherwise it is the index of the device.
```

Comment:
I don't see a --pylon flag. Was it removed?

Reply:
Good catch.  That's why you shouldn't copy-paste code...